draft-thomson-quic-tls.md

```diff
@@ -307,10 +308,24 @@ The following transitions are defined:
 * The server transitions to using 1-RTT keys after sending its Finished message.
   This causes the KEY_PHASE bit to be set to 0 if early data was accepted, and 1
   if the server rejected early data.
+  TODO: I think you're saying that every key transition result in a change
+  to the value of this bit. That makes sense to me. But I'm not understanding
+  the point about rejecting 0-RTT data. Is that the same thing as rejecting the
+  handshake? I'm guessing not. I think I'm not quite able to follow the
+  different values for this bit in all situations. Perhaps it would make sense
+  to show some example with 0-RTT accepted and rejected.
 
 * The client transitions to 1-RTT keys after sending its Finished message.
   Subsequent messages from the client will then have a KEY_PHASE of 0 if 0-RTT
   data was sent, and 1 otherwise.
+  TODO: Hm. This seems to suggest that after the client transitions to forward
+  secure (aka after the handshake is complete), the value of this bit will not
+  be predicatable. This scheme will definitely work, but I think it's a tiny
+  bit undesirable. I alluded to this above, but it would be great if we could
+  get away with not actually needing to send this bit on most packets. Perhaps
+  we can chat about how to do this. Perhaps we could use more bits, but arrage
+  that, say, the value of 0 (which would be implcit if not extra public flags
+  bytes are present) means the first post handshake key.
```

Comment:
I realize that this is a mite icky, but having the bit on every packet is pretty important.  Otherwise, the key transition can be hard/impossible to detect.  QUIC currently does this with trial decryption, but if we want to avoid that, then you need a signal on every packet.  Otherwise when you lose the packet that marks the transition, you lose the signal.

I realize that you could send the signal on every packet until you get some sort of acknowledgment, but that makes the code more complicated.  Besides, there are a few other bits of lower importance in the public flags that can be easily displaced (see above).